aten/src/ATen/native/xnnpack/Common.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#ifdef USE_XNNPACK
+
+#include <xnnpack.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+struct Layout final {
+  // 4D Activation Maps
+  struct Activation4D final {
+    static constexpr size_t Batch = 0u;
+    static constexpr size_t Channels = 1u;
+    static constexpr size_t Height = 2u;
+    static constexpr size_t Width = 3u;
+  };
+
+  // ND Activation Maps
+  struct ActivationND final {
+    static int64_t Batch(const IntArrayRef tensor) {
```

Comment:
Can we have a comment on how this is supposed to work? Why is Batch a product of all but last dimensions? What are semantic meanings of the dimensions here? Are all the tensors that are ActivationND essentially flattened in to 2 dims? It would be good to have a description of the whole approach somewhere.

Reply:
Will add a comment, but basically yeah, a flattened 2D, with the first N-1 dimensions grouped under the "batch" umbrella, and channels as the second dimension.  NHWC -> N * H * W, C.

Reply:
You can do that with NHWC contiguous tensors only. If only nhwc-contiguous tensors are ever passed to xnnpack, that's fine, but it should be described in the design, and there probably should be comments in the code to that effect.

Reply:
To follow up on this issue - added comment to explain what this function does.  Also will assert tensor is contiguous at call site.